lib/sycamore/sycamore/utils/time_trace.py

```diff
@@ -28,34 +79,23 @@ def __exit__(self, exc_type, exc_val, exc_tb):
     def start(self):
         if TimeTrace.fd < 0:
             return
-        self.t0 = time.time_ns()
-        self.r0 = resource.getrusage(self.resource_type)
+        self.imtt = InMemoryTimeTrace()
 
     def end(self):
         if TimeTrace.fd < 0:
             return
-        t1 = time.time_ns()
-        r1 = resource.getrusage(self.resource_type)
+        self.data = self.imtt.end()
```

Comment:
Do we need to store data in the object, or can it just be a local variable?

Reply:
My thinking was that the explicit end function seemed weird without a way to get at the result.  I'm fine with changing if if you don't think that's useful, but in that case, I'd collapse everything into __enter__ and __exit__.

Reply:
Changed.